### ISSUE TITLE
layer name doesn't support Chinese characters

### ISSUE THREAD

**Issue by JadeTao** [chinedufn/psd#4]
```  
  // Read the layer name
    let name_len = cursor.read_u8()?; 
    let name = cursor.read(name_len as u32)?;
    let name = String::from_utf8_lossy(name);
    let name = name.to_string();
```
`圆角矩形 1` in psd turns out to be `Բ�Ǿ��� 1`
name_len: 10
byte:          [212, 178, 189, 199, 190, 216, 208, 206, 32, 49]

plz check out

**Comment by chinedufn** [comment]
Hmmm - my understanding is that Chinese characters are UTF8 encoded.

Any idea why this might not be working? Otherwise I'll have to do some Googling as I'm not too familiar with UTF-8 when it comes to languages besides English - but we definitely don't want things to be broken for others so this is something we want to fix!

**Comment by JadeTao** [comment]
i don't want broken change either.
it seems that problem happens when a *.psd file is read as binary ,the Chinese characters in which are read as wrong utf-8 binary.
so that after parsed to string we get wrong characters.

thanks anyway. i'm looking into it. 
close for now.

**Comment by chinedufn** [comment]
Oh sorry - what I meant was that it looks like this crate is broken if *anyone* has a PSD that uses Chinese characters so we should fix it!

Mind if I re-open this issue until we figure out the problem?

> thanks anyway. i'm looking into it.

Awesome - thank you!

**Comment by tdakkota** [comment]
Cyrillic characters are unsupported too.

**Comment by chinedufn** [comment]
Hmm so I created PSD file with a Chinese layer named `圆角矩形`. In bytes this should be the following 12 bytes `[229, 156, 134, 232, 167, 146, 231, 159, 169, 229, 189, 162]`

I'm not even seeing that sequence of 12 bytes in the Photoshop file.

I google'd for about a half hour or so and couldn't find anything about issues with other languages.

I also tried parsing the layer names in `imagemagick` and got the same `????`

```
identify --version
Version: ImageMagick 7.0.8-27 Q16 x86_64 2019-02-12 https://imagemagick.org
Copyright: © 1999-2019 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP
Delegates (built-in): bzlib freetype heic jng jp2 jpeg lcms ltdl lzma openexr png tiff webp xml zlib
MacBook-Pro:psd chinedufn$ identify -verbose -format "label:%l,geometry:%g\n" tests/fixtures/chinese-layer-name.psd
label:,geometry:1x1+0+0
label:????,geometry:1x1+0+0
```

The data must be in the PSD somewhere since closing and opening the file in Photoshop works - I'm just not sure where it is.

I'm going to stop digging into this for now - but if anyone has a lead on where to find the layer name for different languages I'd be happy to take a look.

**Comment by tdakkota** [comment]
I created PSD file with a Cyrillic layer named `привет`. 
In bytes it is `[239, 240, 232, 226, 229, 242]`. So it's cp1251 encoding(**Note**: PSD file created on Windows machine). 

Also I created PSD file with layer named `圆角矩形`. Photoshop shows layer name correctly, but in bytes it is `[63, 63, 63, 63]`

Code:
```rust
#[test]
fn layer_with_cyrillic_name() {
    let psd = include_bytes!("./fixtures/green-cyrillic-1x1.psd");

    let psd = Psd::from_bytes(psd).unwrap();

    assert_eq!(psd.layers().len(), 1);


    let layer = psd.layer_by_name("привет").unwrap();

    assert_eq!(&layer.rgba().unwrap()[..], &GREEN_PIXEL);
}

#[test]
fn layer_with_chinese_name() {
    let psd = include_bytes!("./fixtures/green-chinese-1x1.psd");

    let psd = Psd::from_bytes(psd).unwrap();

    assert_eq!(psd.layers().len(), 1);

    let layer = psd.layer_by_name("圆角矩形").unwrap();

    assert_eq!(&layer.rgba().unwrap()[..], &GREEN_PIXEL);
}
```

**Comment by chinedufn** [comment]
Ah I think I have a hunch of where to find the data.

According to the spec:

![image](https://user-images.githubusercontent.com/2099811/77851317-7b939200-71a6-11ea-829d-59a2800e39e4.png)

https://www.adobe.com/devnet-apps/photoshop/fileformatashtml/#50577409_pgfId-1049436

The Additional Layer Information section also has the name.

So I think that the `[63, 63, 63, 63]` should just be interpreted as "The layer name isn't here, check the additional layer names section instead"

So I'm thinking that I couldn't find the UTF-8 bytes for the characters because they aren't present in the file as UTF-8, only unicode

**Comment by chinedufn** [comment]
Published as [psd = "0.1.6"](https://crates.io/crates/psd/0.1.6)
